### PR TITLE
Improve install Orchestrator on pre-existing RHDH doc steps for release 1.2

### PR DESCRIPTION
**Pull request by chadcrum**
Includes many improvements to the doc steps for installing Orchestrator on a pre-existing RHDH instance.

Changes:
- Grouped dynamic plugin steps together and app-config steps together 
- Indented configuration examples to more easily copy/paste into RHDH ConfigMaps
- Configuration examples have minimum required config to work when used in a default RHDH instance
- Removed reference of helm values.yaml for Orchestrator Plugins - Replaced with referencing the Orchestrator CRD that contains the default plugin version/integrity for the particular release
- Added NetworkPolicy for sonataflow-infra (Although I believe this will be fixed soon in another PR)
- Overall tried to clarify anything I found confusing when I ran through it

I tested these steps on a pre-existing RHDH 1.2.4 instance using Orchestrator Operator 1.2.1 and everything successfully deployed.

[FLPATH-1819]

### CODE REVIEW

**Comment by chadcrum** [comment]
@masayag @jordigilh @pkliczewski  @jenniferubah

File: docs/release-1.2/existing-rhdh.md [hunk 1]
```diff
@@ -64,121 +91,156 @@ metadata:
...
     .npmrc: |
       @redhat:registry=https://npm.registry.redhat.com
       @<other-scope>:registry=<other-registry>
 ```
 
 ### dynamic-plugins ConfigMap
-This ConfigMap houses the configuration for enabling and configuring dynamic plugins. To incorporate the orchestrator plugins, append the following configuration to the **dynamic-plugins** ConfigMap:
+This ConfigMap houses the configuration for enabling and configuring dynamic plugins in RHDH. 
+
+> Note: Be sure to review [this section](#identify-latest-supported-plugin-versions) to determine the latest supported Orchestrator plugin `package:` and `integrity:` values, and update the dynamic-plugin ConfigMap entries accordingly. The samples in this document may not be the latest.
```
Comment (pkliczewski) [inline docs/release-1.2/existing-rhdh.md]: Would it make sense to provide a link so users know where to check?

File: docs/release-1.2/existing-rhdh.md [hunk 2]
```diff
@@ -64,121 +91,156 @@ metadata:
...
+                  - importName: OrchestratorPage
+                    menuItem:
+                      icon: orchestratorIcon
+                      text: Orchestrator
+                    module: OrchestratorPlugin
+                    path: /orchestrator
 ```
 
-The versions of the plugins may undergo updates, leading to changes in their integrity values. To ensure you are utilizing the latest versions, please consult the values available [here](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/helm-charts/orchestrator/values.yaml). It's imperative to set both the version and integrity values accordingly.
+Additionally, ensure that the `dataIndexService.url` in the above configuration points to the service of the Data Index installed by the Orchestrator Operator.
```
Comment (pkliczewski) [inline docs/release-1.2/existing-rhdh.md]: would it make sense to move this section before the yaml so the users know what to pay attention to as they prepare it?

File: docs/release-1.2/existing-rhdh.md [hunk 3]
```diff
@@ -64,121 +91,156 @@ metadata:
...
 ```bash
 oc get svc -n sonataflow-infra sonataflow-platform-data-index-service -o jsonpath='http://{.metadata.name}.{.metadata.namespace}'
 ```
 
-### app-config ConfigMap
-This ConfigMap used for configuring backstage. Please add/modify to include the following:
-- A static access token (or a different method based on this [doc](https://backstage.io/docs/auth/service-to-service-auth/) to enable the workflows to send notifications to RHDH or to invoke scaffolder actions.
-- Define csp and cors
-
+To include the Notification Plugin:
```
Comment (pkliczewski) [inline docs/release-1.2/existing-rhdh.md]: I think that your comment with `not the latest version` applies here as well. Let's have the warning here as well

**Review by pkliczewski — COMMENTED** [review]
Thank you for your contribution, few minor comments

File: docs/release-1.2/existing-rhdh.md [hunk 4]
```diff
@@ -1,15 +1,36 @@
 # Prerequisites
-- RHDH instance deployed with IDP configured (github, gitlab,...)
-- For using the Orchestrator's [software templates](https://github.com/parodos-dev/workflow-software-templates/tree/v1.2.x), OpenShift Gitops (ArgoCD) and OpenShift Pipelines (Tekton) should be installed and configured in RHDH (to enhance the CI/CD plugins)
-- A secret in RHDH's namespace name `dynamic-plugins-npmrc` that points to the plugins npm registry (details will be provided below)
+- RHDH 1.2 instance deployed with IDP configured (github, gitlab, ...)
+- For using the Orchestrator's [software templates](https://github.com/parodos-dev/workflow-software-templates/tree/v1.2.x), OpenShift Gitops (ArgoCD) and OpenShift Pipelines (Tekton) should be installed and configured in RHDH (to enhance the CI/CD plugins) - [Follow these steps](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/docs/gitops/README.md)
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md:3]: Can you please double check this link 
 `[Follow these steps](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/docs/gitops/README.md)`?
 Maybe we should use `release-1.2` branch instead of `main`

File: docs/release-1.2/existing-rhdh.md [hunk 5]
```diff
@@ -1,15 +1,36 @@
...
-1. Install the orchestrator-operator from the OperatorHub.
+In 1.2, the Orchestrator infrastructure is installed using the Orchestrator Operator.
+1. Install the Orchestrator Operator 1.2 from OperatorHub.
 1. Create orchestrator resource (operand) instance - ensure `rhdhOperator: enabled: False` is set, e.g.
     ```yaml
+    apiVersion: rhdh.redhat.com/v1alpha1
+    kind: Orchestrator
+    metadata:
+      name: orchestrator-sample
+      namespace: rhdh-operator
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md]: Shortly, the RHDH namespace will be configurable (see https://github.com/parodos-dev/orchestrator-helm-operator/pull/320). Do you think we should already consider that in this document?

File: docs/release-1.2/existing-rhdh.md [hunk 6]
```diff
@@ -31,10 +52,16 @@ In 1.2, the Orchestrator infrastructure is being installed using the orchestrato
...
+      tekton:
+        enabled: true
+      argocd:
+        enabled: true
     ```
 1. Verify resources and wait until they are running
-   1. From console run the following command get the necessary wait commands: \
-      `oc describe orchestrator orchestrator-sample -n openshift-operators | grep -A 10 "Run the following commands to wait until the services are ready:"`\
+   1. From the console run the following command in order to get the necessary wait commands: \
+      `oc describe orchestrator orchestrator-sample -n rhdh-operator | grep -A 10 "Run the following commands to wait until the services are ready:"`
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md]: similar question regarding RHDH namespace

File: docs/release-1.2/existing-rhdh.md [hunk 7]
```diff
@@ -64,121 +91,156 @@ metadata:
 EOF
 ```
 The value of `.data.npmrc` points to https://npm.registry.redhat.com.
-For testing RC plugin versions, update to `cmVnaXN0cnk9aHR0cHM6Ly9ucG0uc3RhZ2UucmVnaXN0cnkucmVkaGF0LmNvbQo=` (points to https://npm.stage.registry.redhat.com and can be accessed internally). If there is a need to point to multiple registries, modify the content of the secret's data from:
-```
+For testing RC plugin versions, update to `cmVnaXN0cnk9aHR0cHM6Ly9ucG0uc3RhZ2UucmVnaXN0cnkucmVkaGF0LmNvbQo=` (which points to https://npm.stage.registry.redhat.com and can be accessed internally within Red Hat). If there is a need to point to multiple registries, modify the content of the secret's data from:
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md]: `can be accessed internally within Red Hat`
it brings clarity and I like this change.
I have a doubt about the sentence: this doc is for customers outside our realm. Can we clarify what an external user can do?

File: docs/release-1.2/existing-rhdh.md (hunk 1 above)
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md]: I'm not sure if anchors are rendered properly on our website, can you please check?
In addition, should we refer to a specif version (i.e. `1.2`)?

File: docs/release-1.2/existing-rhdh.md [hunk 8]
```diff
@@ -64,121 +91,156 @@ metadata:
...
-                module: OrchestratorPlugin
-                name: orchestratorIcon
-            dynamicRoutes:
-              - importName: OrchestratorPage
-                menuItem:
-                  icon: orchestratorIcon
-                  text: Orchestrator
-                module: OrchestratorPlugin
-                path: /orchestrator
+      - disabled: false
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md:119]: Was it rendered badly? I mean, is this change intentional?

File: docs/release-1.2/existing-rhdh.md [hunk 9]
```diff
@@ -64,121 +91,156 @@ metadata:
...
+        package: "@redhat/plugin-signals-backend-dynamic@1.2.0"
+        integrity: sha512-DIISzxtjeJ4a9mX3TLcuGcavRHbCtQ5b52wHn+9+uENUL2IDbFoqmB4/9BQASaKIUSFkRKLYpc5doIkrnTVyrA==
 ```
 
-To enable the Notifications plugin, edit the same ConfigMaps.
-For the `dynamic-plugins` ConfigMap add:
+Optionally, include the `plugin-notifications-backend-module-email-dynamic` to fan-out notifications as emails.
+The environment variables below need to be provided to the RHDH instance (Or set the values directly in the ConfigMap).
+See more configuration options for the plugin [here](https://github.com/backstage/backstage/blob/master/plugins/notifications-backend-module-email/config.d.ts).
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md:182]: is `master` branch ok?

File: docs/release-1.2/existing-rhdh.md [hunk 10]
```diff
@@ -64,121 +91,156 @@ metadata:
...
+                    days: 1
 ```
 
-For the `*-app-config` ConfigMap add the database configuration if isn't already provided. It is required for the notifications plugin:
+### app-config ConfigMap
+This ConfigMap is used for configuring backstage. Please add/modify to include the following:
+- `${BACKEND_SECRET}` A static access token to enable the workflows to send notifications to RHDH or to invoke scaffolder actions (Or a different method based on this [doc](https://backstage.io/docs/auth/service-to-service-auth/)).
+- A static access token can be generated with this command `node -p 'require("crypto").randomBytes(24).toString("base64")'`
+- `${RHDH_ROUTE}` can be determined by running `oc get route -A -l app.kubernetes.io/name=backstage`
+- Define csp and cors
```
Comment (gciavarrini) [inline docs/release-1.2/existing-rhdh.md:225]: maybe adding something like "as described in the example below" can clarify this step

**Review by gciavarrini — COMMENTED** [review]
I really like this PR, it brings a lot of clarity to the installation process. 
I have a few questions/comments but great job.
Thanks!

File: docs/release-1.2/existing-rhdh.md (hunk 1 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: `[this section](#identify-latest-supported-plugin-versions)` links to the section in this doc - or are you referring to something else?

File: docs/release-1.2/existing-rhdh.md (hunk 2 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: Makes sense to me - I moved it above. Let me know if that works

File: docs/release-1.2/existing-rhdh.md (hunk 3 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: Sure I copied there as well

File: docs/release-1.2/existing-rhdh.md (hunk 4 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:3]: I agree but we don't seem to have that doc in the `release-1.2` branch. We could back port it and update the link though

File: docs/release-1.2/existing-rhdh.md (hunk 7 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: I had the same feeling about it. How about something like:

> If including plugins from a different NPM registry, the `.data.npmrc` value should be updated with the base64 encoded NPM registry. Example: The Red Hat internal stage NPM registry is `https://npm.stage.registry.redhat.com`, so the value would be updated to `cmVnaXN0cnk9aHR0cHM6Ly9ucG0uc3RhZ2UucmVnaXN0cnkucmVkaGF0LmNvbQo=`  (Note: The Red Hat internal stage registry is only accessible internally within Red Hat)

And also maybe we should just drop the RH stage example and use `https://registry.npmjs.org/` as that would apply more to external users. 

Wdyt?

File: docs/release-1.2/existing-rhdh.md (hunk 4 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md:3]: since there isn't any difference between the versions, I don't think we need to duplicate it.
let's wait with duplicating the gitops folder until there is a need for it.

File: docs/release-1.2/existing-rhdh.md (hunk 1 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: > I'm not sure if anchors are rendered properly on our website, can you please check

Would this actually render in the website (parodos.dev site)? The flow I used to get here is:
- Access website https://www.parodos.dev/1.2/docs/installation/installation-on-existing-rhdh/ 
- Click link in *The installation steps are detailed [here](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/docs/release-1.2/existing-rhdh.md).*
- Which brings the user to github where this doc is rendered md.  The anchor works in github markdown at least

> In addition, should we refer to a specif version (i.e. 1.2)?

I purposely left the version out - the `identify-latest-supported-plugin-versions` section directs the user to reference the Orchestrator CRD installed by the particular Orchestrator Operator. The default plugin versions in the CRD are the supported plugin versions for the Operator release, so they should be used. 

Including @jordigilh and @jenniferubah  on using the CRD for this purpose.

File: docs/release-1.2/existing-rhdh.md (hunk 5 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: I agree we should update this document after the PR is merged, but that isn't necessarily related to the Orchestrator CR namespace.

File: docs/release-1.2/existing-rhdh.md (hunk 8 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:119]: I don't see that on the actual doc. Maybe something with the github diff?

File: docs/release-1.2/existing-rhdh.md (hunk 9 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:182]: This was there originally - I'll direct it to @mareklibra as he actually interacted with the referenced file last.

File: docs/release-1.2/existing-rhdh.md [hunk 11]
```diff
@@ -31,10 +52,16 @@ In 1.2, the Orchestrator infrastructure is being installed using the orchestrato
         serviceNamespace: sonataflow-infra
       rhdhOperator:
         enabled: false
+
+      # Include if using OpenShift Gitops and OpenShift Piplines
```
Comment (masayag) [inline docs/release-1.2/existing-rhdh.md]: I don't think this is needed for the existing RHDH instance.
The assumption is the plugins should already be configured, but even if they aren't, we are not modifying RHDH resource by the chart due to the section mentioned above in this doc:
```
      rhdhOperator:
        enabled: false
```

by that, we won't get to enable the tekton/argoCD plugins:
tekton:
https://github.com/parodos-dev/orchestrator-helm-operator/blob/ebc468f0d7f510824f6741a104afe23413b66db7/helm-charts/orchestrator/templates/rhdh-operator.yaml#L161

argocd:
https://github.com/parodos-dev/orchestrator-helm-operator/blob/ebc468f0d7f510824f6741a104afe23413b66db7/helm-charts/orchestrator/templates/rhdh-operator.yaml#L110
and
https://github.com/parodos-dev/orchestrator-helm-operator/blob/ebc468f0d7f510824f6741a104afe23413b66db7/helm-charts/orchestrator/templates/rhdh-operator.yaml#L166-L177

Instead, we can just ask the user to have those plugins configured as part of the prerequisite.

**Comment by chadcrum** [comment]
cc @y-first

File: docs/release-1.2/existing-rhdh.md (hunk 6 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: RHDH namespace is different than Orchestrator CR namespace.
we don't create resources for RHDH when installing on pre-existing RHDH

File: docs/release-1.2/existing-rhdh.md (hunk 7 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: 100% agree and this looks clear.
The staging one should be mentioned internally in our docs, but not on this github.

File: docs/release-1.2/existing-rhdh.md (hunk 10 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:225]: I updated it - let me know if that works

File: docs/release-1.2/existing-rhdh.md [hunk 12]
```diff
@@ -187,37 +246,8 @@ For the `*-app-config` ConfigMap add the database configuration if isn't already
...
-             sender: sender@mycompany.com
-             replyTo: no-reply@mycompany.com
-             broadcastConfig:
-               receiver: "none"
-             concurrencyLimit: 10
-             cache:
-               ttl:
-                 days: 1
-```
+> Note: `${BACKEND_SECRET}` and `${RHDH_ROUTE}` variables are not by default accessible by backstage, so the values should be used directly in the ConfigMap or made accessible to backstage.
```
Comment (rhdh-orchestrator-test) [inline docs/release-1.2/existing-rhdh.md:249]: nit: Backstage
Also, do we tell the reader how to make the values accessible to Backstage?

File: docs/release-1.2/existing-rhdh.md (hunk 9 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md:182]: I'm okay with master branch. I'm not sure we know how to align the plugin version with its source branch.

**Comment by rhdh-orchestrator-test** [comment]
Note - we will have to update this doc for 1.3 as well.

File: docs/release-1.2/existing-rhdh.md (hunk 12 above)
Comment (y-first) [inline docs/release-1.2/existing-rhdh.md]: nit: Backstage
Also, do we tell the reader how to make the values accessible to Backstage?

**Comment by y-first** [comment]
@chadcrum Do we have a 1.3 version of this doc as well?

File: docs/release-1.2/existing-rhdh.md (hunk 12 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: fixed thanks

> Also, do we tell the reader how to make the values accessible to Backstage?

I purposefully excluded that because IMO the scope of the doc is *the minimal configuration an end user would need to install Orchestrator on a pre-existing, minimally deployed RHDH instance*. 

I think with that scope it shows the end user the basics they can follow to get them working without getting too deep into RHDH configuration, which of course can widely vary. 

I do think that including the variables in the examples could potentially add confusion, even with the various notes. 

wdyt?

Comment (y-first) [reply docs/release-1.2/existing-rhdh.md]: OK so the assumption is that the user is at least minimally familiar with Backstage. That's fine.

Agree about the examples.

**Comment by chadcrum** [comment]
@y-first 
> @chadcrum Do we have a 1.3 version of this doc as well?

We do [here](https://github.com/parodos-dev/orchestrator-helm-operator/blob/main/docs/main/existing-rhdh.md) (main technically) but I think once this gets in for 1.2 we can update the 1.3 version with the changes

File: docs/release-1.2/existing-rhdh.md (hunk 6 above)
Comment (gciavarrini) [reply docs/release-1.2/existing-rhdh.md]: so it shoudn't be `-n rhdh-operator`, or I'm misunderstanding?

File: docs/release-1.2/existing-rhdh.md (hunk 7 above)
Comment (gciavarrini) [reply docs/release-1.2/existing-rhdh.md]: > drop the RH stage example and use https://registry.npmjs.org/
Agree, @batzionb can we do that?

File: docs/release-1.2/existing-rhdh.md (hunk 5 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: Meaning, the namespace in this context should be any namespace.

File: docs/release-1.2/existing-rhdh.md (hunk 6 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: We don't enforce the namespace for this CR, but it should be the same value as the one mentioned in the Orchestrator CD definition in https://github.com/parodos-dev/orchestrator-helm-operator/pull/342/files#diff-3688824233dd051d9cce708de39906213b76573b52c8929522edf83e2ab6b1fbR33

perhaps we can just replace this value with a variable, e.g. `TARGET_NAMESPACE` or any other name to avoid hard-coding it.

File: docs/release-1.2/existing-rhdh.md (hunk 8 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md:119]: Perhaps to allow it being copied and pasted easily without the need to take care of indentation? 
The output of `View file` for this section looks good though.

File: docs/release-1.2/existing-rhdh.md (hunk 12 above)
Comment (masayag) [reply docs/release-1.2/existing-rhdh.md]: +1, this document is for a pre-existing and well configured RHDH that needs to be extended with the Orchestrator plugin configuration and its software templates (assuming they are a good match for the user's tools - I'll leave a comment for that section if it isn't mentioned).

File: docs/release-1.2/existing-rhdh.md [hunk 13]
```diff
@@ -187,37 +246,8 @@ For the `*-app-config` ConfigMap add the database configuration if isn't already
...
-             concurrencyLimit: 10
-             cache:
-               ttl:
-                 days: 1
-```
+> Note: `${BACKEND_SECRET}` and `${RHDH_ROUTE}` variables are not by default accessible by backstage, so the values should be used directly in the ConfigMap or made accessible to backstage. 
+The `${POSTGRES_*}` variables *are* accessible by default, so they can be left in variable form.
 
 ### Import Orchestrator's software templates
 To import the Orchestrator software templates into the catalog via the Backstage UI, follow the instructions outlined in this [document](https://backstage.io/docs/features/software-templates/adding-templates).
```
Comment (masayag) [inline docs/release-1.2/existing-rhdh.md:269]: Let's mention in this context that the templates relies on Github as the git repository system, Quay as the image registry and the GitOps tools are ArgoCD and Tekton, so it will be clear to the user if the template will fit theirs infrastructure or they will have to change it for fit their tools.

File: docs/release-1.2/existing-rhdh.md (hunk 6 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: Ok I updated to $TARGET_NAMESPACE  - let me know if the change looks ok

File: docs/release-1.2/existing-rhdh.md (hunk 7 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: I removed the stage ref and just used registry.npmjs.org, but I can always change to something else

File: docs/release-1.2/existing-rhdh.md (hunk 8 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:119]: I did purposefully adjust for the expected indentation, so maybe that is it.

**Review by masayag — APPROVED** [review]

**Comment by pkliczewski** [comment]
@gciavarrini @y-first are you ok with this PR?

File: docs/release-1.2/existing-rhdh.md (hunk 11 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md]: I removed the section from the Orchestrator CR example and added the argo/tekton plugins to the plugin section. Does that look ok?

File: docs/release-1.2/existing-rhdh.md (hunk 13 above)
Comment (chadcrum) [reply docs/release-1.2/existing-rhdh.md:269]: Added a bulleted list - let me know if that is clear enough

**Review by masayag — APPROVED** [review]

**Comment by pkliczewski** [comment]
@chadcrum please port the changes to main
